generic/tests/downgrade_qcow2_version.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+from autotest.client.shared import error
+
+from virttest import utils_test
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Downgrade qcow2 image version:
+    1) Get the version of the image
+    2) Compare the version with expect version. If they are different,
+    Amend the image with new version
+    3) Check the amend result
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    ver_to = params.get("lower_version_qcow2", "0.10")
+    error_context.context("Downgrade qcow2 image version to '%s'"
+                          % ver_to, logging.info)
+    image = params.get("images").split()[0]
+    t_params = params.object_params(image)
+    qemu_image = qemu_storage.QemuImg(t_params, data_dir.get_data_dir(), image)
+    ver_from = utils_test.get_image_version(qemu_image)
```

Comment:
suggest to compare ver_to and ver_from first, no need to downgrade if they are the same